### PR TITLE
BL808 Watchdog support

### PR DESCRIPTION
**Pull request by grant-olson**
This adds support for the BL808 watchdog. The implementation has a resolution of 1 second and the timeout range can be set from 1 - 2^14 seconds. To test with busybox's watchdog:

* Good watchdog, no reboot: `watchdog -T 10 -t 1 /dev/watchdog -F` This should run just fine.
* Bad watchdog, reboot after a second or two: `watchdog -T 1 -t 10 /dev/watchdog -F`.

### CODE REVIEW

**Comment by Fishwaldo** [comment]
Hi,
your patch seems to have other files. Can you clean up?

 create mode 100644 drivers/watchdog/max_timeout
 create mode 100644 drivers/watchdog/timeout_max

**Comment by grant-olson** [comment]
Must have happened when I rebased after seeing you went to RC6. On it.

**Comment by Fishwaldo** [comment]
Just a FYI, we mostly working off @arm000 tree atm in case you are not aware. 
https://github.com/arm000/linux-bl808

**Review by gamelaster — APPROVED** [review]

**Comment by grant-olson** [comment]
I also sent a PR directly to that repo, and could see some team interactions, but I was a little confused by the branch naming conventions. And I'm a little confused by where your linux patch files come and how they're generated.

In any case I've cleaned up the PR.

File: board/pine64/ox64/patches/linux/0022-Driver-for-BL808-Watchdog.patch [hunk 1]
```diff
@@ -0,0 +1,1438 @@
...
++ziirave_wdt.c:	struct i2c_client *client = to_i2c_client(wdd->parent);
++ziirave_wdt.c:	struct i2c_client *client = to_i2c_client(wdd->parent);
++ziirave_wdt.c:	struct i2c_client *client = to_i2c_client(wdd->parent);
++ziirave_wdt.c:	struct i2c_client *client = to_i2c_client(wdd->parent);
++ziirave_wdt.c:	struct i2c_client *client = to_i2c_client(wdd->parent);
+diff --git a/drivers/watchdog/timeout_max b/drivers/watchdog/timeout_max
+new file mode 100644
+index 000000000000..05af144d06bc
+--- /dev/null
++++ b/drivers/watchdog/timeout_max
```
[the remark this answers is not in the document]
Comment (grant-olson) [reply board/pine64/ox64/patches/linux/0022-Driver-for-BL808-Watchdog.patch]: @Fishwaldo already caught that. PR is updated.

**Comment by Fishwaldo** [comment]
Patches are mainly coming from dyn's tree but a few extra ones relating to opensbi support etc. if you can, join the #ox64-nutcracker channel on the pin64 community. Most of us are cooridinating there.

**Comment by grant-olson** [comment]
Closing. Once this is merged upstream and this project's linux is updated, I'll open a minor PR that just updates the linux_defconfig to add the watchdog.

**Pull request closed without merging**